### DISCUSSION TITLE
What happens, when a worker got disconnected?

### DISCUSSION THREAD

**Discussion by jsiit** [jhuckaby/Cronicle#782] [Q&A]
I use cron jobs for updating dynamic DNS-Entries. So if a client (which a cronicle worker running) get's disconnected by a connection issue, will the worker perform the jobs that it got when the connection was ok? Or will it refuse to do anything until the connection has been restored to the master? What about reporting done actions when the connection re-establishes?

In case that the worker refuses to do anything, I would place a feature-request at this point so have something like a local cache so that the worker performs his jobs (as far as he knows about them) and also caches the results until master is reachable.

**Comment by jsiit** [comment]
Okay...I think I answered the question by myself...The worker stops performing after the master was not reachable.

**So here comes the feature request:** 
Could you please implement some caching-technique to ensure, that the worker can buffer short outages? Also it would be nice, when the worker also caches the results of actions until the master comes back.

**Comment by jhuckaby** [comment]
Upon socket disconnection, the worker keeps performing any active jobs, and when the connection is reestablished, it sends the job updates or job completions to the primary server.

If the worker is offline (unreachable) for longer than the [dead_job_timeout](https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#dead_job_timeout) interval, the job is aborted by the primary server.  If you have a situation where workers go offline or are unreachable for long periods, then increase this value.

Read more here: https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#dead_job_timeout
